music21/interval.py

```diff
@@ -3261,7 +3243,9 @@ def isSkip(self) -> bool:
 
     @property
     def transpositionChangesPitchClass(self) -> bool:
-        return self.simpleName != 'P1'
+        if self.simpleName == 'P1' and float(self.semitones) == float(int(self.semitones)):
+            return False
+        return True
```

Comment:
@mscuthbert I thought `.pitchClass` was the same thing as `.name`, so yeah, transpositionChangesPitchName is in fact what I'm looking for.  But how do I check for that?  `self.simpleName == 'P1'` is True for a DiatonicInterval of 0.5 semitones (Oblique Perfect Unison).  That's why I added "and self.semitones is not fractional" to the condition.

Reply:
I've changed the property name to `transpositionChangesPitchName`, but left the "self.semitones is not fractional" check in place for now, since otherwise it gives wrong answers for P1 +/- half-a-semitone.